mmv1/third_party/terraform/resources/resource_google_service_account.go

```diff
@@ -115,7 +115,22 @@ func resourceGoogleServiceAccountCreate(d *schema.ResourceData, meta interface{}
 		return fmt.Errorf("Error reading service account after creation: %s", err)
 	}
 
-	return resourceGoogleServiceAccountRead(d, meta)
+	err = resourceGoogleServiceAccountRead(d, meta)
+
+	if err != nil {
+		return err
+	}
+
+	// If the id was unset through 404 on the resoource then we assume eventual consistency issue
+	// on part of the api https://cloud.google.com/iam/docs/overview#consistency
+	// Thus we retry ..
+	if d.Id() == "" {
+		d.SetId(sa.Name)
```

Comment:
Why do we set the ID here rather than let it be set in the Read method below?

Reply:
It would've needed to be reset as it was unset by the previous read.

although i've updated the implementation to align with Riley's comment